src/js/module.js

```diff
@@ -8,3 +8,54 @@ Module.noImageDecoding = true;
 Module.noAudioDecoding = true;
 Module.noWasmDecoding = false; // we preload wasm using the built in plugin now
 Module.preloadedWasm = {};
+
+/**
+ *
+ * @param {undefined|(() => string)} stdin
+ * @param {undefined|((text: string) => void)} stdout
+ * @param {undefined|((text: string) => void)} stderr
+ */
+export function setStandardStreams(stdin, stdout, stderr) {
+  if(stdin) {
+    const encoder = new TextEncoder();
+    let input = new Uint8Array(0);
+    let inputIndex = -1; // -1 means that we just returned null
+    function stdinWrapper() {
```

Comment:
Could we add a comment here explaining what we are supposed to be doing (return one the string data one byte one at a time, then return a `null` to indicate that end of input)?

Reply:
Good point, this sort of stuff deserves a comment explaining it/